internal/builder/builder_test.go

```diff
@@ -466,6 +467,28 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 					})
 				})
 
+				when("has repeated buildpacks with the same ID and version", func() {
+					it.Before(func() {
+						subject.AddBuildpack(bp1v1)
+						subject.AddBuildpack(bp1v1)
+					})
+
+					when("order ommits version", func() {
+						it("should de-duplicate identical buildpacks", func() {
+							subject.SetOrder(dist.Order{{
```

Comment:
nit: To make the test reflect the bug a bit more, could we have the buildpack be in two separate groups? 

> There is only one version of the given buildpack on the builder image. However, the buildpack is included in two separate groups. My guess is this is the source of the bug.